### PR TITLE
resizing an image while retaining physical location.

### LINKED ISSUE #4: Resize and scale intensity of image while maintaining aspect ratio
Resize and scale intensity of image, aspect ratio maintained with padding, useful for creating a uniform set of thumbnails from variable sized images, `resize_and_scale_uint8` in [this notebook](https://github.com/InsightSoftwareConsortium/SimpleITK-Notebooks/blob/master/Python/21_Transforms_and_Resampling.ipynb).

### PR DESCRIPTION
**Pull request by zivy**
resolves #4

### CODE REVIEW

**Comment by blowekamp** [comment]
@zivy This does not appear to be a general "resize" method.

**Comment by zivy** [comment]
@blowekamp, you are correct. 

This resize keeps the original **physical** aspect ratio and makes the image isotropic. Function name should reflect this, but I think it is a useful function. Would appreciate any ideas you have for a better function name (it's harder to get the name right than the implementation).

File: SimpleITK/utilities/resize_keep_aspect.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+def resize(
+    image: sitk.Image,
+    new_size: Sequence[int],
+    outside_pixel_value: Union[int, float] = 0,
+) -> sitk.Image:
+    """
+    Resize an image to an arbitrary size, with isotropic pixel spacing.
+
+    Resizing retains the original aspect ratio, with the original image centered
```
Comment (blowekamp) [inline SimpleITK/utilities/resize_keep_aspect.py:14]: aspect ratio of the physical size of the extent of the image

File: SimpleITK/utilities/resize.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+def resize(
+    image: sitk.Image,
+    new_size: Sequence[int],
+    isotropic: bool = True,
+    fill: bool = True,
+    interpolator=sitk.sitkLinear,
+    fill_value: float = 0.0,
+    use_nearest_extrapolator: bool = False,
+    anti_aliasing_sigma: Union[None, float, Sequence[float]] = None,
```
[the remark this answers is not in the document]
Comment (blowekamp) [reply SimpleITK/utilities/resize.py:22]: done

[the remark this answers is not in the document]
Comment (blowekamp) [reply SimpleITK/utilities/resize.py:22]: The resize method is a high level turn key method which automatically configures the itk resample filter to produce a good quality image. Enabling antialiasing by default follows this intention.

[the remark this answers is not in the document]
Comment (zivy) [reply SimpleITK/utilities/resize.py:22]: Got it. Makes sense given the intended turnkey aspect of the function.

**Review by dave3d — APPROVED** [review]
Could use a test with a 3d image.

**Review by dave3d — APPROVED** [review]

**Pull request merged**